packages/smooth_app/lib/data_models/up_to_date_product_provider.dart

```diff
@@ -10,6 +10,16 @@ class UpToDateProductProvider {
   final Map<String, Product> _map = <String, Product>{};
   final Map<String, int> _timestamps = <String, int>{};
 
+  /// Latest downloaded product for a barcode.
+  final Map<String, Product> _latestDownloadedProducts = <String, Product>{};
+
+  /// Barcodes currently displayed in the app.
+  ///
+  /// We need to know which barcodes are "interesting" because we need to cache
+  /// products in memory for instant access. And we should cache only them,
+  /// because we cannot cache all products in memory.
+  final Map<String, int> _interestingBarcodes = <String, int>{};
```

Comment:
Nothing for this PR but overall, I feel like this adds a lot of complexity to the app, maybe we should reevaluate to completely remove this multi tab system 

cc: @teolemon @g123k @AshAman999

Reply:
@M123-dev Actually I've never been a big fan of the multi tab system. But I have enough problems splitting the main PR into smaller ones, I cannot afford to change the whole app UX _en passant_.
That said, my code here goes beyond the multi tab. More specifically, the product list page can have thousands of items but only 10 visible. This is were this "interesting barcodes" feature has a meaning - being careful specifically with displayed barcodes.